src/main/java/nl/tudelft/model/bubble/Bubble.java

```diff
@@ -31,8 +31,8 @@
     private boolean frozen = false;
     private int tickCount = 0;
     private final ResourcesWrapper resources;
-    private final List<Bubble> next;
-    private final boolean goesRightInitially;
+    protected BubbleFactory bubbleFactory = null;
```

Comment:
Which contract is broken? I mean there isn't really a contract that tells Bubble1 to set this field right? 
If you want to make sure it's set, wouldn't it be better to just pass it through the constructor of Bubble?